packages/unlock-js/src/web3Service.ts

```diff
@@ -609,4 +609,83 @@ export default class Web3Service extends UnlockService {
     )
     return price
   }
+
+  /**
+   * Get uniswap price in the out token.
+   * ```ts
+   * const web3Service = new Web3Service(networks)
+   * const price = await web3Service.consultUniswap({
+   *  network: 1,
+   *  tokenInAddress: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+   *  tokenOutAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+   *  value: '1',
+   * })
+   *
+   * console.log(price)
+   * ```
+   */
+  async consultUniswap(options: {
+    tokenInAddress: string
+    tokenOutAddress?: string
+    value: string
+    network?: number
+  }) {
+    const { network, tokenInAddress, value } = options
+    const networkId = network || 1
+    const networkConfig = this.networks[networkId] // By default, use mainnet
+    const tokenOutAddress =
+      options.tokenOutAddress ||
+      networkConfig?.tokens?.find(
+        // By default, use USDC on each network
+        (item: any) => item.symbol === 'USDC'
+      )?.address
+
+    if (!tokenOutAddress) {
+      throw new Error('You need to provide a tokenOutAddress parameter. ')
+    }
+
+    const provider = this.providerForNetwork(networkId)
+
+    if (!networkConfig.unlockAddress) {
+      throw new Error('No unlock contract address found in the network config')
+    }
+
+    const unlockContract = await this.getUnlockContract(
+      networkConfig.unlockAddress!,
+      provider
+    )
+
+    // Get the uniswapContractAddress from unlock contract
+    const uniswapContractAddress = await unlockContract.uniswapOracles(
```

Comment:
Ha no! We can't use the Unlock contract because it may not have all the tokens and we want to support any token that's support by Uniswap, not Unlock. We need to query the Uniswap contract(s) directly.

Reply:
These are public and we can add to the config.

Reply:
(we also save a call!)

Reply:
Okay will directly use the uniswap contracts :D